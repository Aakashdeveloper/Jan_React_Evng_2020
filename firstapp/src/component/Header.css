header{
    background-color:tomato
}

.logo{
    color:#fff;
    text-align:center;
    font-size:30px
}
.uin{
    color:'#fff';
    text-align: center;
    font-size:25px
}