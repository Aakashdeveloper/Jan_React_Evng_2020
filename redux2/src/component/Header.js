import React from 'react';
import { Link} from 'react-router-dom';

const Header = () => {
    return(
        <header>
            <div>
               <h1>hii</h1>
            </div>
        </header>
    )
}

export default Header;