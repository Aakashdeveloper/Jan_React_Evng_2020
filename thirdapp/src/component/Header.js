import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => {
    return(
        <header>
            <Link to="/">Home Page</Link>
        </header>
    )
}

export default Header;