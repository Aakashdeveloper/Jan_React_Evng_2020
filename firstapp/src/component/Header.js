import React, {Component} from 'react';
import './Header.css';



class Header extends Component{
    constructor(){
       super()

       this.state={
           title:'React App',
           keywords:'User text here'
       }
    }

    inputChange(event){
       // console.log(event.target.value)
       this.setState({keywords:event.target.value?event.target.value:'User text here'})
       this.props.userText(event.target.value)
    }

    render(){
        return(
            <header>
                <div className="logo"
                  onClick={()=>{console.log("hiii")}}>
                    {this.state.title}
                </div>

                <center>
                    <input onChange={this.inputChange.bind(this)}/>
                </center>
                <p className="uin">{this.state.keywords}</p>
            </header> 
        )
    }
}


export default Header;