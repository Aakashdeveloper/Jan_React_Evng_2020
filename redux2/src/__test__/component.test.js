import React from 'react';
import Header from '../component/Header';
import { create } from 'react-test-renderer';

describe('SnapShot Testing',() => {
    test('testing header', () => {
        let tree = create(<Header/>)
        expect(tree.toJSON()).toMatchSnapshot()
    })
})