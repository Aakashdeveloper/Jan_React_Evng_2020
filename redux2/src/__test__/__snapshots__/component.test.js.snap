// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Testing testing header 1`] = `
<header>
  <div>
    <h1>
      hii
    </h1>
  </div>
</header>
`;
